engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java

```diff
@@ -2424,4 +2425,28 @@ public int hashCode()
     return result;
   }
 
+  /**
+   * Iterate over all registered {@see Visitor} and invoke visit method
+   * for operator and stream. First visitOperator is called for all operators
+   * in the DAG followed by visitStream is called for all the streams in the
+   * DAG.
+   */
+  public void visitDAG()
+  {
+    Collection<Visitor> visitors = getAttributes().get(VISITORS);
+    if (visitors == null) {
+      return;
+    }
+    for (Visitor visitor : visitors) {
+      LOG.info("Using visitor {}", visitor);
+      visitor.preVisitDAG(this);
+      for (OperatorMeta ometa : operators.values()) {
+        visitor.visitOperator(ometa);
+      }
+      for (StreamMeta smeta : streams.values()) {
```

Comment:
yes, user will have to handle the order in the visitor. If you think the most common cases will need an order then we could provide them from root to leaf.